File: Sota2B.DM/Models/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Sota2B.DM.Models
{
    public class Event
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Reward { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        [JsonIgnore]
        public Achievement? Achievement { get; set; }
        [JsonIgnore]
        public List<UserWasOnEvent> UserWasOnEvent { get; set; } = new List<UserWasOnEvent>();
    }
}

File: Sota2B.Frontend/src/app/data/interfaces/event.ts
import { TuiDay } from "@taiga-ui/cdk";
import { PostAchievement } from "./postAchievement";

export interface IEvent {
    id: number;
    name: string;
    description: string;
    reward: number;
    startDate?: Date;
    endDate?: Date;
    achievement?: PostAchievement;
}
